lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -69,9 +69,10 @@ class Msf::Exploit::Remote::Kerberos::ServiceAuthenticator::Base
   #   @return [String] the encryption key for authentication
   attr_reader :key
 
-  # @!attribute [r] etype
-  #   @return [String] the encryption key type
-  attr_reader :etype
+  # @!attribute [r] offered_etypes
+  #   @return [Array[Integer],Integer,nil] the offered etypes, if not present the default values will be used
```

Comment:
I think long-term this will cause bugs if `offered_etypes` could be an array or integer

```suggestion
  #   @return [Array[Integer],nil] the offered etypes, if not present the default values will be used
```

If we update the callsites to provide an array I think this PR would be good to go

```ruby
    if datastore['NTHASH'].present?
      options[:key] = [datastore['NTHASH']].pack('H*')
      options[:offered_etypes] = [Rex::Proto::Kerberos::Crypto::Encryption::RC4_HMAC]
    end
    if datastore['AESKEY'].present?
      options[:key] = [ datastore['AESKEY'] ].pack('H*')
      if options[:key].size == 32
        options[:offered_etypes] = [Rex::Proto::Kerberos::Crypto::Encryption::AES256]
      else
        options[:offered_etypes] = [Rex::Proto::Kerberos::Crypto::Encryption::AES128]
      end
    end
```

(Or whatever codegolf we want to follow)

Reply:
Dropped support for `offered_types` being an integer in c3254f4e99032544891cceb146d8e143f6a97a1a.